hera_cal/apply_cal.py

```diff
@@ -268,16 +329,31 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
             if flag_file is not None:
                 data_flags[bl] = np.logical_or(data_flags[bl], ext_flags[bl])
         if redundant_solution:
-            from .redcal import get_reds
-            all_reds = get_reds(data.antpos, pols=data.pols(), bl_error_tol=bl_error_tol)
             calibrate_redundant_solution(data, data_flags, new_gains, new_flags, all_reds, old_gains=old_gains,
                                          old_flags=old_flags, gain_convention=gain_convention)
         else:
             calibrate_in_place(data, new_gains, data_flags=data_flags, cal_flags=new_flags,
                                old_gains=old_gains, gain_convention=gain_convention)
-
-        io.update_vis(data_infilename, data_outfilename, filetype_in=filetype_in, filetype_out=filetype_out,
-                      data=data, flags=data_flags, add_to_history=add_to_history, clobber=clobber, **kwargs)
+        if not redundant_average:
+            io.update_vis(data_infilename, data_outfilename, filetype_in=filetype_in, filetype_out=filetype_out,
+                          data=data, flags=data_flags, add_to_history=add_to_history, clobber=clobber, **kwargs)
+        else:
```

Comment:
There's a lot of copy-pasted code from above. Would it make sense to make a helpful function, perhaps a "leading underscore" function with a very short docstring?

Reply:
I could identify about twelve lines that could readily be moved to a function. Given the additional lines required for docstrings and the reduction in readability, i think i'm going to keep this as is.